server/app/mutations/grid_services/common.rb

```diff
@@ -172,7 +172,7 @@ def common_validations
           end
           string :entrypoint
           array :env do
-            string
+            string matches: /^([^=]+)=(.*)/
```

Comment:
`^` matches a character after any newline in the string, not the beginning of the string.